streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1224,17 +1248,20 @@ private int getNextThreadIndex() {
         }
     }
 
-    private long getCacheSizePerThread(final int numStreamThreads) {
+    private long getMemorySizePerThread(final int numStreamThreads, final boolean resizeCache) {
```

Comment:
I think what's originally suggested is a bit more than having a single consolidated function, but also to consolidate the caller: since we are always calling "getMemorySizePerThread" and `resizeThreadCacheOrBufferMemory` for both input buffer and cache size at the same place, we can have a single call rather than two calls. For example, we can just have:

1) a single `resizeThreadCacheAndBufferMemory(final long numThreads)` in which we get the cache-size and input-buffer-size per thread based on the total values of the two and the passed in number of threads, and then let each thread set to the corresponding values directly.

2) the `getMemorySizePerThread` function is then only needed for logging purposes, since otherwise it is now encapsulated inside the `resizeThreadCacheAndBufferMemory` function. We can have just a single `getThreadCacheAndBufferMemoryString()` which returns in the format of "value1/value2" and used in the logging message alone; here we can just get any thread and get its values.

Reply:
@guozhangwang , For the most part this should work. However, in `addStreamThread` , when we need to add a new StreamThread, then at that point, the `createAndAddStreamThread` method expects the newly computed cache size and the new logic for max buffer size. With the consolidated logic and getMemorySizePerThread being converted to return a string to be used for logging purpose only, getting this value to be passed to the constructor might become difficult. Or am I missing something?

Reply:
Could we create the new thread with `0` initial values first, and then call `resizeThreadCacheOrBufferMemory` to set the new values for all threads, including the new thread?

Reply:
Got it. I added the relevant logic.